httpcore/_sync/connection_pool.py

```diff
@@ -294,3 +294,12 @@ def close(self) -> None:
         # Close all connections
         for connection in connections:
             connection.close()
+
+    def get_connection_info(self) -> Dict[str, List[str]]:
+        """Returns dict with ."""
+        stats = {}
+        for origin, connections in self._connections.items():
+            stats[origin_to_url_string(origin)] = [
+                connection.info() for connection in connections
+            ]
+        return stats
```

Comment:
Instead of creating a variable, What do you think about the below code? 

```
return {
    origin_to_url_string(origin): [
        connection.info() for connection in connections
     ]
     for origin, connections in self._connections.items()
}
```

Reply:
Personally I find it less readable that way, not a fan of nested comprehensions 🙂